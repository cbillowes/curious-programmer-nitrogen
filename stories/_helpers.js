import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page