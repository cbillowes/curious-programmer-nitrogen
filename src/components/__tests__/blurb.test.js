import React from "react"
import Blurb from "../blurb"
import { getSnapshot } from "./_helpers"

describe(`Blurb`, () => {
  describe(`should extract content`, () => {
    it(`from plain text`, () => {
      const component = (
        <Blurb>
          The ships hung in the sky in much the same way that bricks don't.
        </Blurb>
      )
      const tree = getSnapshot(component)
      expect(tree).toMatchSnapshot()
    })

    it(`from html elements`, () => {
      const component = (
        <Blurb>
          <h1>Don't panic!</h1>
          <p>Time is an illusion.</p>
          <p>
            Would it save you a lot of time if
            <strong>I just gave up and went mad now?</strong>
          </p>
          <span>
            The ships hung in the sky in much the same way that bricks don't.
          </span>
        </Blurb>
      )
      const tree = getSnapshot(component)
      expect(tree).toMatchSnapshot()
    })
  })

  describe(`should limit words`, () => {
    it(`extracted from plain text`, () => {
      const component = (
        <Blurb wordLimit={6}>
          The ships hung in the sky in much the same way that bricks don't.
        </Blurb>
      )
      const tree = getSnapshot(component)
      expect(tree).toMatchSnapshot()
    })

    it(`extracted from html elements`, () => {
      const component = (
        <Blurb wordLimit={6}>
          <h1>Don't panic!</h1>
          <p>Time is an illusion.</p>
          <p>
            Would it save you a lot of time if
            <strong>I just gave up and went mad now?</strong>
          </p>
          <span>
            The ships hung in the sky in much the same way that bricks don't.
          </span>
        </Blurb>
      )
      const tree = getSnapshot(component)
      expect(tree).toMatchSnapshot()
    })
  })

  describe(`should forcibly limit words`, () => {
    it(`extracted from plain text`, () => {
      const component = (
        <Blurb>
          {Array(300)
            .fill()
            .map((_, i) => ++i)
            .map(i => {
              return i.toString()
            })}
        </Blurb>
      )
      const tree = getSnapshot(component)
      expect(tree).toMatchSnapshot()
    })

    it(`extracted from html elements`, () => {
      const component = (
        <Blurb>
          {Array(300)
            .fill()
            .map((_, i) => ++i)
            .map(i => {
              return <p key={i}>{i.toString()}</p>
            })}
        </Blurb>
      )
      const tree = getSnapshot(component)
      expect(tree).toMatchSnapshot()
    })
  })
})
