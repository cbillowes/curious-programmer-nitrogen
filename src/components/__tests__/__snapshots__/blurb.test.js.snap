// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blurb should extract content from html elements 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Don't panic! Time is an illusion. Would it save you a lot of time if I just gave up and went mad now? The ships hung in the sky in much the same way that bricks don't.",
    }
  }
/>
`;

exports[`Blurb should extract content from plain text 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "The ships hung in the sky in much the same way that bricks don't.",
    }
  }
/>
`;

exports[`Blurb should forcibly limit words extracted from html elements 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50...",
    }
  }
/>
`;

exports[`Blurb should forcibly limit words extracted from plain text 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50...",
    }
  }
/>
`;

exports[`Blurb should limit words extracted from html elements 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "Don't panic! Time is an illusion....",
    }
  }
/>
`;

exports[`Blurb should limit words extracted from plain text 1`] = `
<span
  dangerouslySetInnerHTML={
    Object {
      "__html": "The ships hung in the sky...",
    }
  }
/>
`;
